avocado/core/job.py

```diff
@@ -131,6 +137,8 @@ def __init__(self, config=None):
         #: test was found during resolution.
         self.test_suite = None
 
+        self.suites_configs = suites_configs or [copy(config)]
```

Comment:
We better use `copy.deepcopy` on the configuration, given that we don't want shallow copies to be changing each other's complex data. The following test (feel free to use/adapt) uses the job config and a suite, but the same would happen between suites.

```diff
diff --git a/selftests/unit/test_job.py b/selftests/unit/test_job.py
index 2e3e3b627..f48730b3f 100644
--- a/selftests/unit/test_job.py
+++ b/selftests/unit/test_job.py
@@ -96,6 +96,14 @@ class JobTest(unittest.TestCase):
         self.job = job.Job(config)
         self.assertIsNone(self.job.test_suite)
 
+    def test_job_test_suites_config(self):
+        config = {'run.results_dir': self.tmpdir.name,
+                  'core.show': ['none'],
+                  'run.references': ['/bin/true']}
+        with job.Job(config) as self.job:
+            self.job.suites_configs[0]['run.references'][0] = '/bin/false'
+            self.assertEqual(self.job.config.get('run.references'), ['/bin/true'])
+
     def test_suite_not_started(self):
         suite = TestSuite('empty-suite', {'run.test_runner': 'nrunner'})
         self.assertEqual(suite.status, TestSuiteStatus.RESOLUTION_NOT_STARTED)
```

Reply:
Sure, will change to `deepcopy` in v2. Thanks for the test too.